apitools/org.eclipse.pde.api.tools/META-INF/MANIFEST.MF

```diff
@@ -35,7 +33,10 @@ Export-Package: org.eclipse.pde.api.tools.internal;x-friends:="org.eclipse.pde.a
  org.eclipse.pde.api.tools.internal.util;x-friends:="org.eclipse.pde.api.tools.tests,org.eclipse.pde.api.tools.ui,org.eclipse.pde.api.tools.generator",
  org.eclipse.pde.api.tools.internal.util.profiles;x-internal:=true
 Bundle-RequiredExecutionEnvironment: JavaSE-11
-Import-Package: com.ibm.icu.util
+Import-Package: com.ibm.icu.util,
+ org.objectweb.asm;version="[9.3.0,9.4.0)",
```

Comment:
Please use [9.3.0, 10.0.0) also below

Reply:
as mentioned before I think `[9.3.0, 10.0.0)` is the wrong version range if we extend classes of the `org.objectweb.asm` package.

> Providers implementing an API package must import with a version range that specifies the floor of the version used for compilation and the ceiling of the next minor part.

http://docs.osgi.org/whitepaper/semantic-versioning/060-importer-policy.html

Reply:
... and as mentioned before, this is too narrow range. ASM plans to make breaking API changes in 10 version, not in 9.x, see https://mail.ow2.org/wws/arc/asm/2022-04/msg00000.html

Reply:
@laeubi you are misreading the whitepaper. Here, you are a consumer of the API using a ConsumerType class in the way expected by the API. The API provider will not break you without a major version bump.

Reply:
Do I understand it right, that the only scenario where it's reasonable to use a minor version upper-bound is when you are a 'full' implementer of an API (e.g. like Equinox implements the OSGi API)?
Implementing/extending classes that are intended to be subclassed by consumers does not require the restrict to minor upper-bounds? But a provider of a class intended to be extended/implemented by consumers has to obey the same to SemVer rules when adding new methods to that type like when removing methods from a 'referenced' only API (meaning major-version change in case of addition)?

Reply:
> Here, you are a consumer of the API using a ConsumerType class in the way expected by the API. The API provider will not break you without a major version bump.

I was just a bit curious as the do not use the `@ConsumerType` and `@ProviderType` in their code but the javadoc of `@ConsumerType` clearly states:

> A type is assumed to be [ConsumerType](http://docs.osgi.org/javadoc/r6/annotation/org/osgi/annotation/versioning/ConsumerType.html) if it is not marked either ConsumerType or [ProviderType](http://docs.osgi.org/javadoc/r6/annotation/org/osgi/annotation/versioning/ProviderType.html)